api/src/Responder/Responder/ApiResponder.php

```diff
@@ -137,16 +142,37 @@ protected function collectionWithStatusCode($items, $code = 200)
     }
 
     /**
-     * TODO: Implement paginator() method.
-     * Bind a paginator to a transformer and start building a response.
+     * Build paginated response.
      *
-     * @param Paginator $paginator
+     * @param Collection|array $items
+     * @param null|int $offset
+     * @param null|int $totalCount
      * @param array $parameters
      *
      * @return Response
      */
-    public function paginator(Paginator $paginator, array $parameters = [])
+    public function paginatedCollection($items, $offset = null, $totalCount = null, array $parameters = [])
     {
+        $response = $this->getResponse();
+        $response->headers->set('Content-Type', 'application/json');
+        $offset = is_null($offset)?0:$offset;
+        $totalCount = is_null($totalCount)?'*':$totalCount;
+        $itemCount = count($items);
+        $rangeHeader = $offset.'-'.($itemCount+$offset-1).'/'.$totalCount;
```

Comment:
Can you extract this buildResponseHeader logic out to a separate method to make this method less verbose. Also that way you can separate the ternary defaulting to default property values.